source/base/process_grid.cc

```diff
@@ -216,18 +217,19 @@ namespace Utilities
       if (mpi_process_is_active)
         Cblacs_gridexit(blacs_context);
 
-      MPI_Comm_free(&mpi_communicator_inactive_with_root);
+      if (mpi_communicator_inactive_with_root != MPI_COMM_NULL)
```

Comment:
I think you need to use `MPI_Comm_compare` for this unless `MPI_COMM_NULL` was the initializer of the variable and you never assigned anything to it.

Reply:
@bangerth actually by construction this communicator is either something useful or MPI_COMM_NULL, see https://github.com/dealii/dealii/blob/master/source/base/process_grid.cc#L170-L174
so i think the suggested change is correct.

Reply:
OK, thanks for confirming!